pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -611,14 +619,27 @@ func validateValidatingAdmissionPolicySpec(spec *admissionregistration.Validatin
 			allErrors = append(allErrors, field.Required(fldPath.Child("matchConstraints", "resourceRules"), ""))
 		}
 	}
-	if len(spec.Validations) == 0 {
-		allErrors = append(allErrors, field.Required(fldPath.Child("validations"), ""))
+	if len(spec.Validations) == 0 && len(spec.AuditAnnotations) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath.Child("validations"), "one of validations and auditAnnotations is required"))
+		allErrors = append(allErrors, field.Required(fldPath.Child("auditAnnotations"), "one of validations and auditAnnotations is required"))
 	} else {
 		for i, validation := range spec.Validations {
 			allErrors = append(allErrors, validateValidation(&validation, spec.ParamKind, fldPath.Child("validations").Index(i))...)
 		}
+		if spec.AuditAnnotations != nil {
+			keys := sets.NewString()
+			if len(spec.AuditAnnotations) > maxAuditAnnotations {
+				allErrors = append(allErrors, field.Required(fldPath.Child("auditAnnotations"), fmt.Sprintf("must not have more than %d auditAnnotations", maxAuditAnnotations)))
```

Comment:
Should be `field.Invalid`?

Reply:
Required is documented as being for empty arrays, so I'll use it here.